pkg/apis/serving/v1alpha1/kfservice_status.go

```diff
@@ -14,41 +14,82 @@ limitations under the License.
 package v1alpha1
 
 import (
-	duckv1alpha1 "github.com/knative/pkg/apis/duck/v1alpha1"
+	"github.com/knative/pkg/apis"
 	knservingv1alpha1 "github.com/knative/serving/pkg/apis/serving/v1alpha1"
+	"k8s.io/api/core/v1"
 )
 
 // ConditionType represents a Service condition value
 const (
-	// ServiceConditionRoutesReady is set when the service's underlying
+	// KFServiceConditionRoutesReady is set when the service's underlying
 	// routes have reported readiness.
-	ServiceConditionRoutesReady duckv1alpha1.ConditionType = "RoutesReady"
-	// ServiceConditionDefaultConfigurationsReady is set when the service's underlying
+	KFServiceConditionRoutesReady apis.ConditionType = "RoutesReady"
+	// KFServiceConditionDefaultConfigurationsReady is set when the service's underlying
 	// default configuration have reported readiness.
-	ServiceConditionDefaultConfigurationsReady duckv1alpha1.ConditionType = "DefaultConfigurationReady"
+	KFServiceConditionDefaultConfigurationsReady apis.ConditionType = "DefaultConfigurationReady"
 	// ServiceConditionCanaryConfigurationsReady is set when the service's underlying
 	// canary configuration have reported readiness.
-	ServiceConditionCanaryConfigurationsReady duckv1alpha1.ConditionType = "CanaryConfigurationReady"
+	KFServiceConditionCanaryConfigurationsReady apis.ConditionType = "CanaryConfigurationReady"
 )
 
-var serviceCondSet = duckv1alpha1.NewLivingConditionSet(
-	ServiceConditionDefaultConfigurationsReady,
-	ServiceConditionCanaryConfigurationsReady,
-	ServiceConditionRoutesReady,
+// KFService Ready condition is depending on default configuration and route readiness condition
+// canary configuration readiness condition only present when canary is used and currently does
+// not affect KFService readiness condition.
+var serviceCondSet = apis.NewLivingConditionSet(
```

Comment:
mind spelling out condSet?

K8s status conditions as a list instead of a set always bothered me. I'm glad this is modeled as a set now,